cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/LocalizationUtils.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime;
+
+import co.cask.cdap.internal.app.runtime.distributed.LocalizeResource;
+import com.google.common.io.Files;
+import com.google.common.io.InputSupplier;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
```

Comment:
I don't see any direct dependency on `commons-compress` in app-fabric. If we are using that library, we should have a direct dependency, instead of relaying on some unknown transitive dependency.